pkg/storegateway/bucket.go

```diff
@@ -1627,12 +1627,11 @@ func newBucketBlock(
 		indexHeaderReader: indexHeadReader,
 		// Translate the block's labels and inject the block ID as a label
 		// to allow to match blocks also by ID.
-		relabelLabels: append(labels.FromMap(meta.Thanos.Labels), labels.Label{
+		relabelLabels: labels.Labels{labels.Label{
```

Comment:
I would do a step forward and do some renaming, to further clarify it (given we're not using any blocks "relabelling" at all):
- `relabelLabels` > `blockLabels`
- `matchRelabelLabels()` > `matchLabels()`